components/service/glean/src/main/java/mozilla/components/service/glean/CounterMetricType.kt

```diff
@@ -47,11 +47,6 @@ data class CounterMetricType(
             return
         }
 
-        if (amount <= 0) {
```

Comment:
Let's also get rid of the `// TODO report errors through other special metrics handled by the SDK. See bug 1499761.` comment above. This now has to happen in the storage engine.